ospfd/ospf_lsa.c

```diff
@@ -2112,24 +2114,48 @@ void ospf_external_lsa_rid_change(struct ospf *ospf)
 					continue;
 
 				if (is_prefix_default(
-						(struct prefix_ipv4 *)&ei->p))
+					    (struct prefix_ipv4 *)&ei->p))
 					continue;
 
-				if (!ospf_redistribute_check(ospf, ei, NULL))
-					continue;
+				lsa = ospf_external_info_find_lsa(ospf, &ei->p);
 
 				aggr = ospf_external_aggr_match(ospf, &ei->p);
 				if (aggr) {
+
+					if (!ospf_redistribute_check(ospf, ei,
+								     NULL))
+						continue;
+
 					if (IS_DEBUG_OSPF(lsa, EXTNL_LSA_AGGR))
 						zlog_debug(
 							"Originate Summary LSA after reset/router-ID change");
+
 					/* Here the LSA is originated as new */
 					ospf_originate_summary_lsa(ospf, aggr,
 								   ei);
-				} else if (!ospf_external_lsa_originate(ospf,
-									ei))
-					flog_warn(EC_OSPF_LSA_INSTALL_FAILURE,
-						  "LSA: AS-external-LSA was not originated.");
+				} else if (lsa) {
+					/* LSA need to be  refreshed even there
```

Comment:
fix nit -- `/*LSA needs to be refreshed even if there`

Reply:
Addressed.